tests/SffsFileApi.device.test.nut

```diff
@@ -0,0 +1,272 @@
+
+      // Read and check all written payload data
+      for (local i = 0; i < iterations; i += 10) {
+        local payload = file.read(buffer.len());
+        this.assertDeepEqual(buffer, payload.readstring(buffer.len()));
+      }
+      // all file data readed the position should be EOF
+      this.assertTrue(file.tell() <= file.len());
+      file.close();
+    }
+
+    //
+    // Seek through the file randomly and check payload value
+    function test03_maxFileSizeSeeking() {
+      local file = sffs.open(MAX_SIZE_FILE_FILENAME, "r");
+      local buffer = "0123456789012345678901234567890123456789";
+      local iterations = this.pages * SPIFLASHFILESYSTEM_SECTOR_SIZE / buffer.len() - 10;
+      // random seek
+      for (local i = 0; i < iterations; i += 10) {
+        local randSeek = math.rand() % iterations;
+        randSeek = randSeek * buffer.len();
+        file.seek(randSeek);
+        this.assertEqual(randSeek, file.tell());
+        local payload = file.read(buffer.len());
+        this.assertEqual(buffer, payload.readstring(buffer.len()));
+        this.assertEqual(randSeek + buffer.len(), file.tell());
+      }
+      // move pointer to the end of file
+      file.seek(iterations * buffer.len())
+      // all file data readed the position should be EOF
+      this.assertTrue(file.eof());
+      // Check tell at the end of file
+      this.assertEqual(file.tell(), sffs.fileSize(MAX_SIZE_FILE_FILENAME));
+
+      // check seek and tell at file start position
+      file.seek(0);
+      this.assertEqual(file.tell(), 0);
+      file.close();
+    }
+
+  // seek, tell eof during file writing
+  function test04_writeFileAndSeek() {
+    // remove all files from the previous tests
+    sffs.eraseFiles();
+    // open file for write and seek
+    local file = sffs.open("test.txt", "w");
+    file.write(buffer);
+    // tell return the read-pointer
+    // therefore it should be zero for a newly
+    // created file
+    this.assertEqual(0, file.tell());
+    local payload = file.read(buffer.len());
+    this.assertEqual(buffer.len(), file.tell());
+    this.assertEqual(file.eof(), true);
+    file.seek(0);
+    this.assertEqual(file.tell(), 0);
+    file.write(buffer);
+    file.seek(file.len());
+    this.assertTrue(file.eof());
+    this.assertEqual(file.tell(), buffer.len() * 2);
+    file.seek(file.len() - 1);
+    this.assertTrue(!file.eof());
+
+    file.close();
+    // there no way to read data from closed file
+    this.assertEqual(file.seek(0).tell(), 0);
+    // try to read data from closed file
+    payload = file.read(buffer.len());
+    this.assertEqual(file.tell(), buffer.len());
+    this.assertEqual(file.seek(file.len()).tell(), file.len());
+    this.assertTrue(file.eof());
+  }
+
+
+  // seek, tell eof after file close
+  function test05_fileClose() {
+    local file = sffs.open("test.txt", "r");
+    file.close();
+
+    this.assertEqual(file.tell(), 0);
```

Comment:
The previous one was for a new file, but this one if for reading of an existing file.

And one more interesting aspect of this test is that we have close previous test with file seek position equal to the eof.  And we double check that a new instance of the opened file has a correct seek position.